src/hooks/maps/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { useRef, useEffect, EffectCallback } from "react";
+import { createCustomEqual } from "fast-equals";
+import { isLatLngLiteral } from "@googlemaps/typescript-guards";
+
+const deepCompareEqualsForMaps = createCustomEqual(
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  (deepEqual) => (a, b) => {
+    if (
+      isLatLngLiteral(a) ||
+      a instanceof google.maps.LatLng ||
+      isLatLngLiteral(b) ||
+      b instanceof google.maps.LatLng
+    ) {
+      return new google.maps.LatLng(a).equals(new google.maps.LatLng(b));
+    }
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return deepEqual(a, b);
+  }
+);
+
+function useDeepCompareMemoize(value) {
+  const ref = useRef();
+
+  if (!deepCompareEqualsForMaps(value, ref.current)) {
+    ref.current = value;
+  }
+
+  return ref.current;
+}
+
+export function useDeepCompareEffectForMaps(
+  callback: EffectCallback,
+  dependencies: any[]
```

Comment:
Here `useDeepCompareEffectForMaps` is doing something similar to `useEffect`. It has a callback and then a dependency array. So, if you want to change this `any[]` maybe you could use same type use in `useEffect` called `DependencyList` imported from React
![image](https://user-images.githubusercontent.com/40902023/200851933-e5d02d78-442a-49cc-a17d-d8a78aa88ed5.png)

Reply:
Perfect, thank you!